src/ProfileEndpoints.jl

```diff
@@ -21,11 +21,16 @@ using Serialization: serialize
 #
 #----------------------------------------------------------
 
-function _http_response(binary_data, filename)
-    return HTTP.Response(200, [
-        "Content-Type" => "application/octet-stream"
-        "Content-Disposition" => "attachment; filename=$(repr(filename))"
-    ], body = binary_data)
+function _http_create_response(binary_data, filename, should_inline_data = true)
```

Comment:
:) having one of the arguments be `nothing` sometimes strikes me as an indication that this should really be two separate methods. Does that sound right to you too?